index.xml

```diff
@@ -303,6 +324,26 @@ x-example: Example header with some whitespace.
     </figure>
    </t>
 
+        <t>
+If you want to add an optiona date or expires to the signature, you
```

Comment:
@liamdennehy do you like it? This should work with `fetch` too.

Reply:
@dlongley this should address your comment when `Date` is either:

- forbidden
- not mandatory

Reply:
Yeah, I like this approach. We might want `created` instead of `date` though (bikeshed).

Reply:
Since this changes the signature algorithm a bit, I suggest it goes out in the new spec under a new "algorithm" mechanism... that is, rsa-sha1 (or whatever ancient thing we're using right now) should be changed to what we're doing in the Linked Data Proof/Signature we're doing at W3C... that is, we give the signing string construction algorithm a yearly identifier so we don't break future upgradability. rsa-sha1 will remain a "legacy" mechanism that we deprecate over time. 

Agree with where this PR is going... nicely done @ioggstream!

Reply:
@dlongley I inherited the `date` and `expires` from https://tools.ietf.org/id/draft-yasskin-http-origin-signed-responses-05.html#rfc.section.3.1 (which are probably going to register `Signature` into iana).

@msporny thanks for your feedback! It's a long time :)

1- I think we should plan for a versioned release of http-signatures. If `Signatures` will be registered by https://github.com/WICG/webpackage/blob/master/draft-yasskin-http-origin-signed-responses.md#iana-considerations   we could move to a new header with a more precise semantic. 

2- In the simple case, we can just add a new optional parameter containing the http-signature version, like it's done in other specs. I found this interesting reading on the subject https://tools.ietf.org/id/draft-thomson-postel-was-wrong-03.html.